src/swapd/swap_state.rs

```diff
@@ -114,10 +124,287 @@ pub enum SwapCheckpointType {
     CheckpointAlicePreBuy,
 }
 
+pub enum Awaiting {
+    // Msg
+    TakerP2pMakerCommit,
+    MakerTakerP2pRevealProof,
+    BobP2pRevealAlice,
+    MakerTakerP2pReveal,
+    AliceP2pCoreArb,
+    BobP2pRefundSigs,
+    AliceP2pBuySig,
+    // Ctl
+    TakerCtlTakeSwap,
+    MakerCtlRevealProof,
+    MakerTakerCtlFundingUpdated,
+    MakerCtlMakeSwap,
+    BobCtlCoreArb,
+    AliceCtlRefundSigs,
+    BobCtlBuySig,
+    BobCtlTxLock,
+    AliceBobCtlTx,
+    AliceBobCtlSweepMonero,
+    AliceBobCtlTerminate,
+    AliceBobCtlSweepBitcoin,
+    AliceBobCtlRpcAbortSimple,
+    BobCtlRpcAbortBob,
+    AliceBobCtlRpcAbortBlocked,
+    AliceBobRpcGetInfo,
```

Comment:
The `AliceBob` naming is a bit confusing to me. It took a second for me to understand that these are just the ones that are shared between them. Is it possible to just use the shared predicates without the `AliceBob` prefix and similarly for the state evaluation and transition functions?